### PR TITLE
Manual Provider Subnet Discovery

### PR DESCRIPTION
**Pull request by manadart**
## Description of change

This change allows us to turn on subnet discovery during update of link-layer devices detected by machine agents.

If the model is for a _manual_ cloud, incoming link-layer device data causes subnets that we have not yet observed to be added to state.

A following patch will turn on provider space support and ensure that constraints/bindings can be honoured in manual clouds.

## QA steps

Assuming access to a machine with IP 172.16.99.2:
- `juju bootstrap manual/ubuntu@172.16.99.2 manual-subnets --no-gui --debug`.
- `juju switch controller`.
- `juju subnets` should show:
```yaml
subnets:
  172.16.99.0/24:
    type: ipv4
    status: in-use
    space: alpha
    zones: []
```

## Documentation changes

Once space support is turned on, I will ensure appropriate documentation changes.

## Bug reference

N/A

### CODE REVIEW

**Comment by manadart** [comment]
[Populate Subnets for Manual Provider](https://trello.com/c/KqX5m2vq/2565-populate-subnets-for-manual-provider)

File: state/subnets.go [hunk 1]
```diff
@@ -438,11 +451,14 @@ func (st *State) addSubnetOps(id string, args network.SubnetInfo) (subnetDoc, []
 	if !unique {
 		return subnetDoc{}, nil, errors.AlreadyExistsf("subnet %q", args.CIDR)
 	}
+
+	// Unless explicitly placed, new subnets go into the alpha space.
+	// TODO (manadart 2020-08-12): We should determine the model's configured
+	// default space and put it there instead.
```
Comment (achilleasa) [inline state/subnets.go:457]: Should we print a warning or something so we don't forget about this?

Comment (manadart) [reply state/subnets.go:457]: I have a ticket to do this, and also to address https://bugs.launchpad.net/bugs/1883164.
Hoping to get this in for 2.9 so we can keep ratcheting towards our pure space concept.

**Comment by manadart** [comment]
$$merge$$

**Pull request merged**